sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -3013,6 +3013,74 @@ SIREPO.app.directive('emailLoginConfirm', function(requestSender, $route) {
     };
 });
 
+SIREPO.app.directive('ldapLogin', function(requestSender, errorService) {
+    return {
+        restrict: 'A',
+        scope: {
+            email: '@',
+            password: '@',
+        },
+        template: `
+            <form class="form-horizontal" autocomplete="off" novalidate>
+              <div class="form-group">
+                <div class="col-sm-offset-2 col-sm-10">
+                  <p>Enter your LDAP login</p>
+                </div>
+              </div>
+              <div class="form-group">
+                <label class="col-sm-2 control-label">Email</label>
+                <div class="col-sm-10">
+                  <input type="text" value='' maxlength="256" class="form-control" data-ng-model="email"/>
+                </div>
+                <label class="col-sm-2 control-label">Password</label>
+                <div class="col-sm-10">
+                  <input type="text" value='' maxlength="256" class="form-control" data-ng-model="password"/>
+                </div>
+              </div>
+              <div class="form-group">
+                <div class="col-sm-offset-2 col-sm-10">
+                  <div data-disable-after-click="">
+                    <button data-ng-click="login()" class="btn btn-primary">Continue</button>
+                  </div>
+                  <div class="sr-input-warning" data-ng-show="showWarning">{{ warningText }}</div>
+                  <p class="help-block">By signing up for Sirepo you agree to Sirepo\'s <a href="en/privacy.html">privacy policy</a> and <a href="en/terms.html">terms and conditions</a>, and to receive informational and marketing communications from RadiaSoft. You may unsubscribe at any time.</p>
+                </div>
+              </div>
+            </form>
+        `,
+        controller: function($scope) {
+            function handleResponse(data) {
+                if (data.state == 'ok') {
```

Comment:
I think data.state should always be ok here. If it isn't, it should throw an error.

Reply:
Changing to display 'contact support@sirepo.com' error seen in [emailLogin](https://github.com/radiasoft/sirepo/blob/b93ebec93ed5f83cce9f20980ff7be02c8e78556/sirepo/package_data/static/js/sirepo-components.js#L2963)